feast_component/component.py

```diff
@@ -23,34 +23,65 @@
 from tfx.components.example_gen import component
-from tfx.components.example_gen import utils
 from tfx.dsl.components.base import executor_spec
-from tfx.examples.custom_components.presto_example_gen.presto_component import executor
 from tfx.proto import example_gen_pb2
 
-from protos import FeastConfig
+from . import executor
 
 
-class FeastExampleGen(component.QueryBasedExampleGen):    
+class FeastExampleGen(component.QueryBasedExampleGen):
 
     EXECUTOR_SPEC = executor_spec.BeamExecutorSpec(executor.Executor)
 
-    def __init__(self,
-                 conn_config: FeastConfig.FeastConnConfig,
-                 query: Optional[str] = None,
-                 input_config: Optional[example_gen_pb2.Input] = None,
-                 output_config: Optional[example_gen_pb2.Output] = None):
+    def __init__(
+        self,
+        repo_config: feast.RepoConfig,
+        entity_query: Optional[str] = None,
+        features: Optional[List[str]] = None,
+        feature_service: Optional[str] = None,
+        **kwargs
+    ):
+        """FeastExampleGen is a way for loading offline features. For now we support BQ.
 
-        packed_custom_config = example_gen_pb2.CustomConfig()
-        packed_custom_config.custom_config.Pack(conn_config)
+        Args:
+            repo_config (feast.RepoConfig): Feast repo configuration object
+            entity_query (Optional[str], optional): Query used to obtain the entity dataframe. Defaults to None.
+            features (Optional[List[str]], optional): List of features to retrieve as part of this job. Defaults to None.
+            feature_service (Optional[str], optional): Feature service identifier used to retrieve historical features.
+                Attributes features and feature_service can't be set at the same time. Defaults to None.
+            **kwargs: kwargs used in QueryBasedExampleGen
+        """
+        if features and feature_service:
+            raise RuntimeError(
+                "Exactly one of feature_view_ref and feature_refs should be set."
+            )
 
-        output_config = output_config or utils.make_default_output_config(
-            input_config)
+        # Serialize repo config into a YAML to pass it to the executor
+        # ToDo: Potentially better would be to actually push the YAML to some pipeline path and load it back from executor to avoid having too much data in the entrypoint!
+        repo_yaml = None
+        with tempfile.TemporaryDirectory() as t:
+            repo_config.write_to_path(pathlib.Path(t))
+            with open(os.path.join(t, "feature_store.yaml")) as f:
+                repo_yaml = f.read()
 
-        super().__init__(
-            input_config=input_config,
-            output_config=output_config,
-            custom_config=packed_custom_config)
\ No newline at end of file
+        # Store configuration as part of a protobuf struct and pack inside custom_config
+        custom_config = Struct()
```

Comment:
We should probably define a `FeastCustomConfig` object to encapsulate these values instead of using `Struct` directly.

Reply:
Imo, better to just use a struct. Having a custom proto means having it all setup for a proto. Given this is an internal detail, I don't think it matters that much.